src/components/Dashboard/ThisWeek/NewsletterMenu/MenuItems/LogNewsLetter/LogNewsletter.test.tsx

```diff
@@ -1,15 +1,187 @@
+              <LogNewsletter
+                accountListId={accountListId}
+                handleClose={handleClose}
+              />
+            </GqlMockedProvider>
+          </SnackbarProvider>
+        </AppProvider>
+      </MuiPickersUtilsProvider>,
+    );
+    userEvent.click(getByRole('button', { name: 'Close' }));
+    expect(handleClose).toHaveBeenCalled();
+  });
+
+  describe('Logging Newsletter', () => {
+    it('Logs Physical Newsletter', async () => {
+      const { getByRole, getByText, findByText } = render(
+        <MuiPickersUtilsProvider utils={LuxonUtils}>
+          <AppProvider
+            initialState={{
+              user: {
+                id: 'user-1',
+                firstName: 'John',
+                lastName: 'Smith',
+              } as User,
+            }}
+          >
+            <SnackbarProvider>
+              <GqlMockedProvider<CreateTaskMutation>>
+                <LogNewsletter
+                  accountListId={accountListId}
+                  handleClose={handleClose}
+                />
+              </GqlMockedProvider>
+            </SnackbarProvider>
+          </AppProvider>
+        </MuiPickersUtilsProvider>,
+      );
+      expect(getByText('Log Newsletter')).toBeInTheDocument();
+
+      userEvent.click(getByText('Save'));
+      expect(await findByText('Field is required')).toBeInTheDocument();
+
+      userEvent.type(getByRole('textbox', { name: 'Subject' }), accountListId);
+      await waitFor(() => expect(getByText('Save')).not.toBeDisabled());
+      userEvent.click(getByText('Save'));
+      await waitFor(() => expect(handleClose).toHaveBeenCalled());
+    });
+
+    it('Logs Email Newsletter', async () => {
+      const { getByRole, getByText, findByText } = render(
+        <MuiPickersUtilsProvider utils={LuxonUtils}>
+          <AppProvider
+            initialState={{
+              user: {
+                id: 'user-1',
+                firstName: 'John',
+                lastName: 'Smith',
+              } as User,
+            }}
+          >
+            <SnackbarProvider>
+              <GqlMockedProvider<CreateTaskMutation>>
+                <LogNewsletter
+                  accountListId={accountListId}
+                  handleClose={handleClose}
+                />
+              </GqlMockedProvider>
+            </SnackbarProvider>
+          </AppProvider>
+        </MuiPickersUtilsProvider>,
+      );
+      expect(getByText('Log Newsletter')).toBeInTheDocument();
+
+      userEvent.click(getByText('Save'));
+      expect(await findByText('Field is required')).toBeInTheDocument();
+
+      userEvent.type(getByRole('textbox', { name: 'Subject' }), accountListId);
+      await waitFor(() => expect(getByText('Save')).not.toBeDisabled());
+      userEvent.click(getByRole('radio', { name: 'Newsletter - Email' }));
+      userEvent.click(getByText('Save'));
+      await waitFor(() => expect(handleClose).toHaveBeenCalled());
```

Comment:
@OzzieOrca Do we have a way to test if the mutation actually ran? I know we had something like that in MH, I believe it was a helper method like ```toHaveBeenMutatedWith```. This does the job as ```handleClose``` should only run after the mutation fires but let me know what you think.

Reply:
There's https://github.com/SurveyMonkey/graphql-ergonomock#ergonomockedprovider#:~:text=Finally,%20you%20can%20spy%20on%20executed%20operations%20via%20the%20onCall%20prop.

Reply:
Yep, that seems to work 🙂 I may wait on using it for now as I'm not sure the best way to utilize it. The response object it provides is a giant blob of json and I don't think it would be great to have to do an inline snapshot on that... let me know if you have any ideas. Do you think checking if the ```operation.variables``` match what we expected them to be would be worth the effort?

Reply:
Ya might be worth checking that the right variables were sent. How hard does it seem?